sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala

```diff
@@ -445,4 +447,15 @@ object ShimLoader extends Logging {
   def newExplainPlan(): ExplainPlanBase = {
     newInstanceOf[ExplainPlanBase]("com.nvidia.spark.rapids.ExplainPlanImpl")
   }
+
+  def newHiveProvider(): HiveProvider= {
+    newInstanceOf[HiveProvider]("org.apache.spark.sql.hive.rapids.HiveProviderImpl")
+  }
+
+  /**
+   * Singleton Avro Provider
+   */
+  lazy val avroProvider: AvroProvider = ShimLoader.newInstanceOf[AvroProvider](
```

Comment:
I'd rather have only methods in these trait, and not introduce any constraints, and let the caller decide how to use it. Can we change this to:
```
def newAvroProvider(): AvroProvider = { 
    ...
}
```
for consistency.

Reply:
Done